thirdeye/thirdeye-frontend/app/pods/components/rootcause-trend/component.js

```diff
@@ -0,0 +1,389 @@
+import { computed, observer } from '@ember/object';
+import { later } from '@ember/runloop';
+import Component from '@ember/component';
+import {
+  toCurrentUrn,
+  toBaselineUrn,
+  toOffsetUrn,
+  hasPrefix,
+  filterPrefix,
+  toMetricLabel,
+  isInverse,
+  toColorDirection,
+  makeSortable
+} from 'thirdeye-frontend/utils/rca-utils';
+import {
+  humanizeChange,
+  humanizeFloat,
+  humanizeScore
+} from 'thirdeye-frontend/utils/utils';
+import moment from 'moment';
+import _ from 'lodash';
+
+const ROOTCAUSE_TREND_MAX_COLUMNS = 12;
+
+export default Component.extend({
+  classNames: ['rootcause-metrics'],
+
+  //
+  // external properties
+  //
+
+  /**
+   * Entities cache
+   * @type {object}
+   */
+  entities: null,
+
+  /**
+   * Metric aggregates
+   * @type {object}
+   */
+  timeseries: null,
+
+  /**
+   * Investigation context
+   * @type {object}
+   */
+  context: null,
+
+  /**
+   * User-selected urns
+   * @type {Set}
+   */
+  selectedUrns: null,
+
+  /**
+   * Callback on metric selection
+   * @type {function}
+   */
+  onSelection: null, // function (Set, state)
+
+  /**
+   * loading status for component
+   * @type {boolean}
+   */
+  isLoading: false,
+
+  //
+  // internal properties
+  //
+
+  /**
+   * Display flag for table. Hack for ember models table only re-rendering columns on init()
+   * @type {bool}
+   */
+  showTable: true,
+
+  /**
+   * Hack for ember models table only re-rendering columns on init()
+   */
+  _columnObserver: observer('columns', function () {
```

Comment:
Hi Alex. Not sure i understand the need for this observer. We should avoid using observer when possible due to it's synchronous nature. Can we not use an `init()` hook here?  I can't seem to find `_columnObserver` usage. 
https://guides.emberjs.com/v2.6.0/components/the-component-lifecycle/

Also why not have ` showTable: true,` be false by default vs setting to `this.set('showTable', false);`.

```
  init() {
    this._super(...arguments);
...
  },
```

Reply:
the table component does not re-render if we change the columns property after it has been initialized. My workaround for now is to destroy and re-create with this flag. If there is a better solution, I'd be glad to use it instead

Reply:
there is a better solution. :-)